lib/record_store/provider/cloudflare/client.rb

```diff
@@ -0,0 +1,92 @@
+require 'net/http'
+
+module Cloudflare
+  class Client
+    attr_reader :config
+
+    def initialize(config = Cloudflare::Configuration.new)
+      @config = config
+    end
+
+    def get(endpoint, params = {})
+      uri = build_uri(endpoint, params)
+
+      response = request(:get, uri)
+      Cloudflare::Response.new(response)
+    end
+
+    def post(endpoint, body = nil)
+      uri = build_uri(endpoint)
+
+      response = request(:post, uri, body: body)
+      Cloudflare::Response.new(response)
+    end
+
+    def put(endpoint, body = nil)
+      uri = build_uri(endpoint)
+
+      response = request(:put, uri, body: body)
+      Cloudflare::Response.new(response)
+    end
+
+    def patch(endpoint, body = nil)
+      uri = build_uri(endpoint)
+
+      response = request(:patch, uri, body: body)
+      Cloudflare::Response.new(response)
+    end
+
+    def delete(endpoint)
+      uri = build_uri(endpoint)
+
+      response = request(:delete, uri)
+      Cloudflare::Response.new(response)
+    end
+
+    def request(method, uri, body: nil)
```

Comment:
I see that you fetch errors in `response.rb` but you'll only see those errors after you run `Cloudflare::Response.new(response)`. Do we not need to handle errors that could be raised from `Net::HTTP`?